### PR TITLE
 Feature/380/rsvp-prompt-for-users-who-have-rvsp'd

### PR DESCRIPTION
**Pull request by ABCodez**
Fixes issue #380.
(un-RVSP functionality not handled in this PR)

image

### CODE REVIEW

File: src/pages/Home/Home.page.tsx [hunk 1]
```diff
@@ -76,6 +97,17 @@ const HomePage = () => {
...
                 </Card>
 
+                <Card title="RSVP Status" className="xl:col-span-5">
+                    <span className="flex flex-col gap-2">
+                        <p className="text-[#333] text-sm">
+                            RSVP status: <b>{rsvpStatus}</b>
+                        </p>
+                        <button className="p-2 bg-white rounded-md">
+                            Not able to make it?
+                        </button>
```
Comment (juancwu) [inline src/pages/Home/Home.page.tsx:107]: We are in the process of making the waitlist, so I wouldn't put this here just yet.

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/pages/Home/Home.page.tsx (hunk 1 above)
Comment (ABCodez) [reply src/pages/Home/Home.page.tsx]: Sounds good, removed in the newest commit

File: functions/src/index.ts [hunk 2]
```diff
@@ -385,6 +385,32 @@ export const logEvent = functions.https.onCall((data, context) => {
...
+            message: "RSVP verified.",
+        };
+    } else {
+        return {
+            status: 200,
+            verified: false,
+            message: "RSVP not verified.",
+        };
+    }
+});
```
Comment (juancwu) [inline functions/src/index.ts:412]: we don't really need this function. We have their rsvp status in the FE already.

File: src/pages/Home/Home.page.tsx [hunk 3]
```diff
@@ -44,6 +46,25 @@ const Sponsors = sponsors.map((sponsor, i) => {
...
+            try {
+                const result = await fetchRSVPStatus();
+                console.log("RSVP status:", result.data);
+                setRsvpStatus(result.data ? "RSVP'd" : "Not RSVP'd");
+            } catch (error) {
+                console.error("Error fetching RSVP status:", error);
+                setRsvpStatus("Error fetching status");
+            }
+        };
+
```
Comment (juancwu) [inline src/pages/Home/Home.page.tsx:64]: we can use `currentUser.rsvpVerified` from the auth provider.

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/pages/Home/Home.page.tsx (hunk 3 above)
Comment (juancwu) [reply src/pages/Home/Home.page.tsx:64]: You can check out `routes.provider.tsx` how i use it

File: functions/src/index.ts (hunk 2 above)
Comment (ABCodez) [reply functions/src/index.ts]: updated

File: src/pages/Home/Home.page.tsx (hunk 3 above)
Comment (ABCodez) [reply src/pages/Home/Home.page.tsx]: updated in newest commit

**Review by juancwu — DISMISSED** [review]

**Pull request merged**